rotkehlchen/chain/ethereum/modules/yearn/vaultsv2.py

```diff
@@ -215,15 +217,19 @@ def get_vaults_history(
                     to_block=to_block,
                 )
                 continue
+            self.database.add_yearn_vaults_events(address, events)
+            events_mapping[address] = db_events + events
 
+        for address in addresses:
+            if address in events_mapping:
+                all_events = events_mapping[address]
+            else:
+                all_events = self.database.get_all_yearn_vaults_v2_events(address=address)
```

Comment:
Does this respect the block restrictions of the `get_vaults_history()` function? I don't think it does. You also used this before at line 206.

Where in this function are the events limited to the given block range?

Reply:
Also true sir :tophat: